addOns/quickstart/src/main/java/org/zaproxy/zap/extension/quickstart/AttackThread.java

```diff
@@ -49,8 +49,14 @@ public enum Progress {
     private TraditionalSpider traditionalSpider;
     private PlugableSpider plugableSpider;
     private boolean stopAttack = false;
-    private boolean useStdSpider;
+    private boolean pauseAttack = false;
+    private boolean resumeAttack = false;
```

Comment:
Why do you have two variables for this? Either it's paused or not. Trying to keep track of 4 possible values is just a headache

Reply:
I was trying to find a good trigger for unpausing and I wasn't sure how to best have it trigger.  Right now there is a check to see if it is stopped which triggers the stopscan feature and I needed things for pause and resume scan.  The issue though was if I just check if pauseAttack is false, I risk causing trouble in the code where it will try to resume scan.

Reply:
Wouldn’t paused false be the same as resumed true?

Reply:
So, starting out, both are false.  This means it is just starting.  But once the pause button is pressed, pause is true and resume is false and ideally pausing happens.  Then, when resume happens, pause is false but resume is true which ideally should cause the app to resume instead of using start.